common/lib/capa/capa/util.py

```diff
@@ -65,24 +65,18 @@ def compare_with_tolerance(student_complex, instructor_complex, tolerance=defaul
         # `tolerance` both equal to infinity. Then, below we would have
         # `inf <= inf` which is a fail. Instead, compare directly.
         return student_complex == instructor_complex
+
+    if isinstance(student_complex, float) and isinstance(instructor_complex, float):
```

Comment:
@adampalay the line at
https://github.com/edx/edx-platform/blob/6be35e0fc1418fb58fbc8d362bdde5ca9ac2dd82/common/lib/capa/capa/responsetypes.py#L1255

is producing the correct_float of "complex" type (instructor_complex) so the when the user press the check button it will never go at this code however for float types such as we are using in test_util.py its very clean solution